token/js/src/actions/amountToUiAmount.ts

```diff
@@ -28,3 +31,189 @@ export async function amountToUiAmount(
+        amount,
+        mintInfo.decimals,
+        currentTime,
+        interestBearingMintConfigState.lastUpdateTimestamp,
+        interestBearingMintConfigState.initializationTimestamp,
+        interestBearingMintConfigState.preUpdateAverageRate,
+        interestBearingMintConfigState.currentRate
+    );
+}
+
+/**
+ * Convert an amount with interest back to the original amount without interest
+ * This implements the same logic as the CPI instruction available in /token/program-2022/src/extension/interest_bearing_mint/mod.rs
+ * 
+ * @param uiAmount                  UI Amount (principle plus continuously compounding interest) to be converted back to original principle
+ * @param decimals                  Number of decimals for the mint
+ * @param currentTimestamp          Current timestamp in seconds
+ * @param lastUpdateTimestamp       Last time the interest rate was updated in seconds
+ * @param initializationTimestamp   Time the interest bearing extension was initialized in seconds
+ * @param preUpdateAverageRate      Interest rate in basis points (hundredths of a percent) before the last update
+ * @param currentRate              Current interest rate in basis points
+ * 
+ * @return Original amount (principle) without interest
+ */
+
+export function uiAmountToAmountWithoutSimulation(
+    uiAmount: string,
+    decimals: number,
+    currentTimestamp: number, // in seconds
+    lastUpdateTimestamp: number,
+    initializationTimestamp: number,
+    preUpdateAverageRate: number,
+    currentRate: number,
+): bigint {
+    Decimal.set({ toExpPos: 24, toExpNeg: -24 })
+    const uiAmountDecimal = new Decimal(uiAmount);
+    const decimalsFactor = new Decimal(10).pow(decimals);
+    const uiAmountScaled = uiAmountDecimal.mul(decimalsFactor);
+   
+    // Calculate pre-update exponent
+    const preUpdateTimespan = new Decimal(lastUpdateTimestamp).minus(initializationTimestamp);
+    const preUpdateNumerator = new Decimal(preUpdateAverageRate).times(preUpdateTimespan);
+    const preUpdateExponent = preUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const preUpdateExp = preUpdateExponent.exp();
+
+    // Calculate post-update exponent
+    const postUpdateTimespan = new Decimal(currentTimestamp).minus(lastUpdateTimestamp);
+    const postUpdateNumerator = new Decimal(currentRate).times(postUpdateTimespan);
+    const postUpdateExponent = postUpdateNumerator.div(new Decimal(SECONDS_PER_YEAR).times(ONE_IN_BASIS_POINTS));
+    const postUpdateExp = postUpdateExponent.exp();
+
+    // Calculate total scale
+    const totalScale = preUpdateExp.times(postUpdateExp);
+
+    // Calculate original principle by dividing the UI amount (principle + interest) by the total scale
+    const originalPrinciple = uiAmountScaled.div(totalScale);
+    return BigInt(originalPrinciple.trunc().toString()); 
+}
+
+/**
+ * Convert a UI amount with interest back to the original UI amount without interest
```

Comment:
nit: since this works more generally, how about this?
```suggestion
 * Convert a UI amount back to the raw amount
```